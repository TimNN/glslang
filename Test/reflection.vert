#version 440 core

uniform nameless {
    vec3 anonMember1;
    vec4 anonDeadMember2;
    vec4 anonMember3;
};

uniform named {
    vec3 deadMember1;
    vec4 member2;
    vec4 member3;
} ablock;

uniform namelessdead {
    int a;
};

uniform namedDead {
    int b;
} bblock;

struct TS {
    int a;
    int dead;
};

uniform TS s;

uniform float uf1;
uniform float uf2;
uniform float ufDead3;
uniform float ufDead4;

const bool control = true;

void deadFunction()
{
    vec3 v3 = ablock.deadMember1;
    vec4 v = anonDeadMember2;
    float f = ufDead4;
}

void liveFunction2()
{
    vec3 v = anonMember1;
    float f = uf1;
}

void liveFunction1()
{
    liveFunction2();
    float f = uf2;
    vec4 v = ablock.member3;
}

void main()
{
    liveFunction1();
    liveFunction2();

    if (! control)
        deadFunction();

    float f;

    if (control) {
        liveFunction2();
        f = anonMember3.z;
        f = s.a;
    } else
        f = ufDead3;
}
