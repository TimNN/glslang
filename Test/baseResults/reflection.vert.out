reflection.vert
Warning, version 440 is not yet complete; some version-specific features are present, but many are missing.



Linked vertex stage:



Uniform reflection:
0:anonMember3: offset 32, type 35666, arraySize 1, index 0
1:s.a: offset -1, type 35666, arraySize 1, index -1
2:anonMember1: offset 0, type 35666, arraySize 1, index 0
3:uf1: offset -1, type 35666, arraySize 1, index -1
4:uf2: offset -1, type 35666, arraySize 1, index -1
5:ablock.member3: offset 32, type 35666, arraySize 1, index 1

Uniform block reflection:
0: nameless: offset -1, type -1, arraySize 1, index -1
1: ablock: offset -1, type -1, arraySize 1, index -1

Live names
ablock: 1
ablock.member3: 5
anonMember1: 2
anonMember3: 0
liveFunction1(: -1
liveFunction2(: -1
nameless: 0
s.a: 1
uf1: 3
uf2: 4

